lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1276,40 +1276,31 @@ public function __toString() {
 	 *
 	 * @since 6.2.0
 	 *
-	 * @param array|string $query {
+	 * @param array $query {
 	 *     Which tag name to find, having which class.
 	 *
 	 *     @type string|null $tag_name     Which tag to find, or `null` for "any tag."
 	 *     @type string|null $class_name   Tag must contain this class name to match.
 	 * }
 	 */
 	private function parse_query( $query ) {
-		if ( null !== $query && $query === $this->last_query ) {
-			return;
+		if ( null !== $query ) {
+			if ( ! is_array( $query ) || $query === $this->last_query ) {
+				return;
+			}
```

Comment:
the nesting catches my attention.

I don't think we need to change this code at all because we only set the query parameters if we have an array with the given keys. `isset()` is inherently safe to call.

it would be different if we did something to alert a dev of passing a non-array value, but right now there's no difference between passing a non-array and passing an array with the wrong keys.

Reply:
Well, good point! I just reverted this part